### PR TITLE
Add multi-release support, bump SDK to 2.11.3.121 for xenial and 2.9.3.43 for trusty

### PR DESCRIPTION
**Pull request by Baycken**
* Fixed all catkin lint -W2 errors and warnings. 
Only significant change is change the how `download_flycap` is run from cmake: Execution bit of the script is removed and the cmake invoke python to run the script instead of running the script as executable. This is done to resolve warning about not installed executable.

* Use cmake magic to choose which version on flycap to use for xenial and trusty thus allow both release to be used. 

* Bump the SDK version for xenial to `2.3.11.121`  and for trusty to `2.3.9.43`.

### CODE REVIEW

File: pointgrey_camera_driver/CMakeLists.txt [hunk 1]
```diff
@@ -100,6 +101,5 @@ if (CATKIN_ENABLE_TESTING)
   roslaunch_add_file_check(launch/camera.launch)
   roslaunch_add_file_check(launch/stereo.launch)
 
-  find_package(roslint REQUIRED)
```
Comment (mikepurvis) [inline pointgrey_camera_driver/CMakeLists.txt]: This line needs to remain, or the code linter macro won't work.

Comment (Baycken) [inline pointgrey_camera_driver/CMakeLists.txt]: I removed it from the roslint find_package for catkin anf added it back

Comment (mikepurvis) [inline pointgrey_camera_driver/CMakeLists.txt]: The issue is that `roslint` is only a `test_depend`, so it needs to only be looked for inside the `if(CATKIN_ENABLE_TESTING)` block.

**Comment by mikepurvis** [comment]
Thanks very much for looking at this— I know there have been a number of reports around 16.04, particularly in #73, a previous PR along similar lines.

It looks like you have a bunch of `__cxx11` linker errors (indicating an STL ABI mismatch) occurring in the Travis environment, which is Ubuntu Trusty + GCC 4.8.4. Since we _do_ get a clean build from the buildfarm (running Xenial with GCC 5.3.1), I think our options are:

- Drop support for Indigo, at least in the master branch. Either eliminate the Travis job, or switch it to use a containerized Xenial.
- Make the build able to detect which STL is present, and download the correct version of FlyCapture, thus maintaining some semblance of Trusty/Indigo support on the master branch.

Thoughts? @konradb3 @ironmig @tedchina

**Comment by ironmig** [comment]
I'd like to see the indigo compatibility maintained as there are still several years left in indigo support and I'm sure there's a lot of robots out there running indigo with pointgrey cameras still (my lab has at least one). Perhaps create an indigo-devel branch like many big ROS project have adopted and occasionally backport changes if they build (while maintaining this diff in the SDK version of the cmake). What do you think about this?

**Comment by Baycken** [comment]
I added version selection for Ubuntu releases by using code name, I think this way both indigo/trusty and kinetic/xenial can be supported

**Comment by Baycken** [comment]
@mikepurvis, how does this look now?

**Comment by mikepurvis** [comment]
Looks terrific. Thanks for your patience and persistence in getting this through.

**Pull request merged**